### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by dyazdani**
Changed error handling to set status codes near error and removed that from `app.ts`. Tests were adjusted and all work:

Screen Shot 2023-12-29 at 11 20 31

### CODE REVIEW

File: src/utils/authentication.ts [hunk 1]
```diff
@@ -27,7 +27,14 @@ const authenticateJWT = (req: Request, res: Response, next: NextFunction) => {
                 req.user = null
             }
             next();
-        } catch (e) {
+        } catch (e: any) {
```
Comment (jvaneyken) [inline src/utils/authentication.ts:30]: I don't think the error should be typed as any here

**Review by jvaneyken — CHANGES_REQUESTED** [review]
Overall Looks good. Just one minor change requested

File: src/utils/authentication.ts (hunk 1 above)
Comment (dyazdani) [reply src/utils/authentication.ts:30]: I agree, but apparently a catch block only allows any or unknown typing (or at least that's what my VS code linter said). Maybe there is a way to expect an Error type without a catch block?

Comment (dyazdani) [reply src/utils/authentication.ts:30]: Fixed it.

**Review by jvaneyken — APPROVED** [review]
LGTM

**Pull request merged**